junipervpn/vpn.py

```diff
@@ -386,8 +386,10 @@ def main():
         sys.exit(1)
 
     jvpn = juniper_vpn(args)
-    signal.signal(signal.SIGINT, jvpn.stop)
-    jvpn.run()
+    try:
+        jvpn.run()
+    finally:
```

Comment:
Cool, this LGTM - I've just done a quick test and this does have the correct behaviour on Ctrl-C (i.e. stop *is* always called). This is a much nicer way to do it that the signal handler :)